src/AliasRecord.php

```diff
@@ -257,6 +267,8 @@ protected function remapOptionTable()
     {
         return [
             'user' => 'remote-user',
+            # service is used by Docker Compose transport.
+            'service' => 'remote-host',
```

Comment:
Why does `service` map to `remote-host`?  `service` will have a value such as `drupal`; it wouldn't be right to specify `--remote=host=drupal`, would it?

Reply:
I think that RedispatchHook is another area we need to fix. It is using `--remote-host` as an indicator (from the legacy code), but then when it actually redispatches, it asks the alias manager for the 'self' alias, and passes that on to SiteProcess.

I think instead that we should ask for the self alias earlier, and replace the check for the `remote-host` option with some other logic. Maybe the SiteProcess library needs to provide an API to indicate that a given alias would be handled by a transport. (Maybe remove LocalTransport from consideration in this API, and use that as a fallback in SiteProcess::getTransport() if nothing matches.

Reply:
I think that the remapOptionTable uses dot-access notation, and therefore could work; however, I am still -1 on using `remote-host` here.  I think that we should use `docker.service` and `docker.exec.options` for clarity (rather than just `service` and `exec.options`).